### PR TITLE
Missing headers won't cause key not found exception

### LINKED ISSUE #2: SagaAudit plugin throws a key not found exception when processing a message that does not contain the `TimeSent`, `OriginatingMachine` and `OriginatingEndpoint`
When an endpoint that contains the SagaAudit plugin processes a message that arrives at the endpoint and the message does not contain the `TimeSent`, `OriginatingMachine` and/or `OriginatingEndpoint` headers, even though the endpoint is able to successfully process the message, the saga audit plugin throws a KeyNotFound exception causing the message to end up in the error queue.
## 
### Steps to repro:
1. Add a saga
2. Use queue explorer to create a new message, paste in the body of a message
### Exception

```
System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at ServiceControl.Plugin.SagaAudit.CaptureSagaStateBehavior.AuditSaga(Saga saga, IncomingContext context) in c:\BuildAgent\work\5d8e2b68f32aa08f\src\ServiceControl.Plugin.SagaAudit\CaptureSagaStateBehavior.cs:line 64

```
### Solution

Cater for the fact that the header collection can be empty on line 64-67:
https://github.com/Particular/ServiceControl.Plugins.v5/blob/47454d30b0e79c91f1a2dbae0c6e54be3a8769a3/src/ServiceControl.Plugin.SagaAudit/CaptureSagaStateBehavior.cs#L64-67

**Related Googlegroups post:**
https://groups.google.com/forum/#!topicsearchin/particularsoftware/saga$20audit$20plugin/particularsoftware/PoHI4Zj98kI

### PR DESCRIPTION
**Pull request by indualagarsamy**
If a message does not contain the OriginatingMachine, OriginatingEndpoint and TimeSent headers, message will still be audited. Fixes #2

### CODE REVIEW

**Comment by indualagarsamy** [comment]
@SimonCropp - can you review this change? I started out trying to add a unit test the whole behavior, but couldn't as I can't new up my own IncomingContext. Do you still see value in the test i added? 
cc / @andreasohlund @gbiellem @SeanFeldman

**Comment by SimonCropp** [comment]
@indualagarsamy make the constructor public. for now construct it with reflection

File: src/ServiceControl.Plugin.Nsb5.SagaAudit.Tests/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
+﻿using System.Reflection;
```
Comment (SimonCropp) [inline src/ServiceControl.Plugin.Nsb5.SagaAudit.Tests/Properties/AssemblyInfo.cs]: remove the assembly info. it adds no value to a test assembly

File: src/ServiceControl.Plugin.Nsb5.SagaAudit/Messages/SagaChangeInitiator.cs [hunk 2]
```diff
@@ -7,7 +7,7 @@ class SagaChangeInitiator
         public string InitiatingMessageId { get; set; }
         public string MessageType { get; set; }
         public bool IsSagaTimeoutMessage { get; set; }
-        public DateTime TimeSent { get; set; }
+        public DateTime? TimeSent { get; set; }
```
Comment (SimonCropp) [inline src/ServiceControl.Plugin.Nsb5.SagaAudit/Messages/SagaChangeInitiator.cs:10]: is this a breaking change?

Comment (indualagarsamy) [inline src/ServiceControl.Plugin.Nsb5.SagaAudit/Messages/SagaChangeInitiator.cs:10]: I always return DateTime.MinValue (as opposed to null) - see Line94. So SC won't break. 
I wanted to be able to return a null at some point, when SC can accept a null. 
Does that make sense?

Comment (SimonCropp) [inline src/ServiceControl.Plugin.Nsb5.SagaAudit/Messages/SagaChangeInitiator.cs:10]: so SI, SP and SC will all correctly interpret minvalue as a "non value"?

Comment (indualagarsamy) [inline src/ServiceControl.Plugin.Nsb5.SagaAudit/Messages/SagaChangeInitiator.cs:10]: @SimonCropp - In SI, not sure at the moment, I was going to check with @distantcam, in SP I think @gbiellem has fixed it. I'll double check. If not what's your proposal. To roll out SP and SI first enabling for decent display, before we roll out this change? Or do you have anything else in mind?

**Pull request merged**